api/src/services/validation-service.ts

```diff
@@ -310,7 +310,8 @@ export class ValidationService extends DBService {
       throw SubmissionErrorFromMessageType(SUBMISSION_MESSAGE_TYPE.INVALID_MEDIA);
     }
 
-    const csvState: ICsvState[] = file.isContentValid(parser);
+    const csvState: ICsvState[] = [...file.isWorkbookValid(parser), ...file.isContentValid(parser)];
```

Comment:
I wonder if we need to spread `isWorkbookValid` here? Won't it have written any errors into the worksheets already, which will be accounted for in the return from `isContentValid`?

I guess the only issue is if there are no `isContentValid` rules defined, but there are `isWorkbookValid` rules. This is where merging them becomes necessary maybe. Assuming I'm following this all correctly.

Reply:
Actually, what we probably need, is instead of returning anything from these 2 functions. We should just call them, and then get the csvStates afterwards from some separate function. Then it doesn't matter what validation rules are being run (if any), we can get the csvStates once at the end.

Reply:
Just saw your comment here (https://github.com/bcgov/biohubbc/pull/890/files/0ccd6f87af42e88d9f5dcc906b6958c535a95c6b#r1044892317). Possibly my comments here are irrelevant.

Reply:
Ah you're right. Merging them like this would case duplicates, since the CSV states from `isWorkbookValid` will get caried forward. Short of combing through each list of `ICsvState` and taking the state of a given CSV to be derived from one or the other, maybe this calls for the creation of `IWorkbookState`? 🤥

Reply:
> Actually, what we probably need, is instead of returning anything from these 2 functions. We should just call them, and then get the csvStates afterwards from some separate function. Then it doesn't matter what validation rules are being run (if any), we can get the csvStates once at the end.

Ahh yeah good call. That's what we're doing inside of the `isWorkbookValid`, anyway.

Reply:
> Actually, what we probably need, is instead of returning anything from these 2 functions. We should just call them, and then get the csvStates afterwards from some separate function. Then it doesn't matter what validation rules are being run (if any), we can get the csvStates once at the end.

This what I ended up opting for.